### PR TITLE
WIP: Externalize hooked domain values based on their sort

### PR DESCRIPTION
**Pull request by u--**
###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`
- [ ] Style conformance: `stylish-haskell`

---

### CODE REVIEW

File: src/main/haskell/kore/src/Kore/IndexedModule/Resolvers.hs [hunk 1]
```diff
@@ -248,21 +249,46 @@ resolveSort m sortId =
...
+    -> Text
+    -> Sort Object
+    -> Either (Error e) (Id Object)
+resolveHookWithSort indexedModule builtinName builtinSort =
+    resolveHookHandler builtinName $
+        filter
+        (\name ->
+            elem
+            builtinSort
+            ((\s -> applicationSortsResult s : applicationSortsOperands s)
```
[the remark this answers is not in the document]
Comment (u--) [reply src/main/haskell/kore/src/Kore/IndexedModule/Resolvers.hs]: I thought so too, but then there's stuff like `LIST.get` ...

File: src/main/haskell/kore/src/Kore/IndexedModule/Resolvers.hs [hunk 2]
```diff
@@ -248,21 +249,46 @@ resolveSort m sortId =
...
-            koreFail
-                ("Builtin '" ++ Text.unpack builtinName
-                    ++ "' is hooked to multiple identifiers: "
-                    ++ List.intercalate ", " (squotes . getIdForError <$> hookIds)
-                )
-          where
-            squotes str = "'" ++ str ++ "'"
+    resolveHookHandler builtinName $ resolveHooks indexedModule builtinName
+
+resolveHookWithSort
```
[the remark this answers is not in the document]
Comment (u--) [reply src/main/haskell/kore/src/Kore/IndexedModule/Resolvers.hs]: Oh. I just meant I wasn't going to duplicate every single `lookup` function in `Builtin.List, Set, Map`. I think it's fine for this to exist.

File: src/main/haskell/kore/src/Kore/IndexedModule/Resolvers.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (u--) [reply src/main/haskell/kore/src/Kore/IndexedModule/Resolvers.hs]: Ohh. I see what you mean. In that case, though we *do* need both versions of `resolve`, in order to resolve `LIST.get`.

File: src/test/resources/expected/test-hooks-10.kore.golden [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            , moduleSentences =
+                [ ObjectSentence (SentenceHookSentence (SentenceHookedSort SentenceSort
+                    { sentenceSortName = (Id "S" AstLocationNone) :: Id Object
+                    , sentenceSortParameters = []
+                    , sentenceSortAttributes =
+                        Attributes 
+                            [ Fix (UnifiedPattern (UnifiedObject (Rotate31 (ApplicationPattern Application
+                                { applicationSymbolOrAlias =
+                                    SymbolOrAlias
+                                        { symbolOrAliasConstructor = (Id "hook" AstLocationNone) :: Id Object
```
Comment (virgil-serbanuta) [inline src/test/resources/expected/test-hooks-10.kore.golden]: These lines are way too long.

Comment (ttuegel) [reply src/test/resources/expected/test-hooks-10.kore.golden]: This is a golden file; what can we do about it?

**Pull request merged**